frame/balances/src/lib.rs

```diff
@@ -214,6 +215,12 @@ pub mod pallet {
 		/// The maximum number of locks that should exist on an account.
 		/// Not strictly enforced, but used for weight estimation.
 		type MaxLocks: Get<u32>;
+
+		/// The maximum number of named reserves that can exist on an account.
+		type MaxReserves: Get<u32>;
+
+		/// The id type for named reserves.
+		type ReserveIdentifier: Parameter + Member + MaxEncodedLen + Ord + Copy;
```

Comment:
Is there a reason to make `ReserveIdentifier` configurable?

Sounds like it will be a lot harder to use.

`LockIdentifier` is the same and defined in the primitives.

Reply:
Requested by @gavofyork 

https://github.com/paritytech/substrate/pull/7778#discussion_r642944170
https://github.com/paritytech/substrate/pull/7778#discussion_r643042191

I think the goal is make transition to fungible named holds in future easier.

Reply:
The goal is to ensure that the number of values of `ReserveIdentifier` (and thus the number of reserves than can possibly be in place at once) is capped at compile time through Rust's type system.

Reply:
> This will obviate the need for `TooManyReserves` error type and `MaxReserves` parameter.

May I ask why the `MaxReserves` parameter was still necessary after all?